lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -0,0 +1,110 @@
+require 'fastlane/action'
+require 'digest/sha1'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      S3_UPLOADED_FILE_PATH = :S3_UPLOADED_FILE_PATH
+    end
+
+    class UploadToS3Action < Action
+      def self.run(params)
+        file_path = params[:file]
+        file_name = File.basename(file_path)
+
+        bucket = params[:bucket]
+        key = params[:key]
+
+        key = file_name if key.nil?
```

Comment:
```suggestion
        key = params[:key] || file_name
```
(This use of `||` is the Ruby equivalent of `??` in Swift)